dlib/dnn/layers.h

```diff
@@ -3985,31 +4079,28 @@ namespace dlib
 
 // ----------------------------------------------------------------------------------------
 
+    template <unsigned long s_mode_>
     class softmax_
     {
     public:
-        softmax_() 
-        {
-        }
+        softmax_() {}
 
         template <typename SUBNET>
-        void setup (const SUBNET& /*sub*/)
-        {
-        }
+        void setup(const SUBNET& /*sub*/) {}
 
         void forward_inplace(const tensor& input, tensor& output)
         {
-            tt::softmax(output, input);
-        } 
+            tt::softmax(output, input, static_cast<operation_mode>(s_mode_));
```

Comment:
```suggestion
            tt::softmax(output, input, s_mode_);
```
E.g. then this is just this.   There shouldn't be any `static_cast<operation_mode>()` happening.

Reply:
Thanks for updating the code accordingly.

Reply:
No problem :)